[Detection Engine] Investigation: Testing whether es_archiver can cause duplicate documents

rylnd opened the pull request:
This integration test is executed 100 times, with the single-document archive being loaded and unloaded before/after each test. If this test fails, it will eliminate the detection engine as a culprit.

rylnd committed:
Testing whether es_archiver can cause duplicate documents

This integration test is executed 100 times, with the single-document
archive being loaded and unloaded before/after each test. If this test
fails, it will eliminate the detection engine as a culprit.

rylnd commented:
200 runs (with 100 executions each, for a total of 20_000 executions): https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8349

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8349](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8349)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 160/200 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd committed:
Merge branch 'main' into rylnd/break_es_archiver

 Conflicts:
	x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/index.ts

rylnd committed:
Manually enable APM and set full sampling rate

I don't know if APM is disabled by default in my scenario, but this will
ensure it is.

rylnd commented:
Attempting to trigger another build with APM manually enabled: https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8561

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8561](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8561)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 71/100 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd committed:
Attempt to enable APM for the flaky test runner

I think the values in this config will be respected, as per this thread:
https://elastic.slack.com/archives/C05UT5PP1EF/p1741788732650719, but
we'll find out.

rylnd committed:
whoopsie, typo

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8577](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8577)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 0/10 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd committed:
Adding a few more overrides for good measure

At this point I'm just seeing what sticks.

rylnd committed:
Adjust mutex env var

After looking more closely at how these env variables interact, I think
that the context propagation var needs to explicitly be false in order
to allow reporting; I had assumed that this wasn't needed originally.

This also removes the duplicated definitions in a substep of the
pipeline.

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8600](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8600)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 1/10 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/317954)
* Commit: 2fbc0a902642589c227e0740db211d918ef6141a

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/317954#0197f655-daab-4566-acf2-6ad41d046a2a)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/317954","number":317954,"commit":"2fbc0a902642589c227e0740db211d918ef6141a"}],"number":317954}
buildkite-pr-comment-->

rylnd committed:
Remove raw `console.log`s to appease linter

Now that we're running an actual build, we can't get to the tests until
it lints 🙄

rylnd commented:
/ci

rylnd committed:
Add APM configuration for detection engine FTRs

This is copied liberally from the APM performance FTRs.

rylnd committed:
Revert most of our intermediate config changes

So that this all happens at the base config, instead.

rylnd commented:
New FTR build cribbing APM's FTR config: https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8661#01980fd1-6b52-41f2-bdd6-e264f39cae28

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8661](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8661)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 9/10 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd committed:
Logging test config

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8662](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8662)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 7/10 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd committed:
One more attempt to enable APM on flaky-test-runner

* Adds a delay at the end of the run to allow APM to report
* Adds server args for telemetryOptIn and labels

I assumed telemetry was opt out, but perhaps this was the missing piece.

rylnd committed:
Debug: log registered APM config

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8690](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8690)
[✅] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 10/10 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd committed:
Attempt to unset the APM API key for flaky tests

We set a default key from vault, but my guess is that this is
conflicting with the server/token being used to send to the performance
APM server.

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8691](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8691)
[✅] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 10/10 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd committed:
Attempt to override setting of APM API key

I suspect that setting the pipeline env var to `undefined` might not
override an existing value, there, so I'm trying to remove any setting
of that API key at all (in addition to unsetting the server URL, which
_seems_ to be getting overridden elsewhere, to the perf server, but at
this point I have no confidence in anything.

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8692](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8692)
[✅] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 10/10 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd commented:
Okay, with [this change](https://github.com/elastic/kibana/pull/223043/commits/706c8d965a752414a67ba167f2535ff8cd1bbeee) I was able to unset the default APM API key, which was causing our reporting to the perf server to be rejected. I'm now able to see my data there, but:

1. I'm not sure if it's the data that was asked for
2. The data doesn't yet include a failure

I'm addressing the second point with [a new 50x build](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8695), and will attempt to clarify the first point with the others involved.

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8695](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8695)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 25/50 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd commented:
`labels.ciBuildJobId: "01981533-3610-44fa-b238-a3a3cf351589"` is a failing job from the previous build.

rylnd committed:
Apply @dgieselaar's instrumentation of FTR

This adds APM around our functional test runner, which should allow us
to collect data on what's happening with elasticsearchduring our
(failing) test runs.

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8738](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8738)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 0/20 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F223043%2Fhead)

rylnd commented:
Closing this for now, in favor of the more productive (and more straightforward) https://github.com/elastic/kibana/pull/228850.
